### DISCUSSION TITLE
Laravel Vapor Brand Logo Path?

### DISCUSSION THREAD

**Discussion by dmason30** [laravel/nova-issues#4041] [Q&A] [answered]
- Laravel Version: 9.7.0
- Nova Version: 4.2.2
- PHP Version: 8.1.4

### Description:
Locally this config loads the logo correctly:
```php
'brand' => [
    'logo' => public_path('images/logo.svg') // Shows empty logo on deploy
    // 'logo' => realpath(__DIR__.'/../public/images/logo.svg') // Shows default Laravel Nova logo on deploy,

    'colors' => [
        // ...
    ],
],
```

However, when it is deployed with vapor the public directory does not exist and it is unable to read the file and we end up with a blank logo in the portal:

<img width="407" alt="image" src="https://user-images.githubusercontent.com/20278756/164000999-80723b5b-433d-4ad9-b86d-5a804d2d39b1.png">

The svg is uploaded to s3 with cloudront? What is the recommended way to configure the logo when using vapor?

**Comment by crynobone** [comment]
You should be able to do the following:

https://twitter.com/MikeVILT/status/1514581480240029710

Also instead of passing string directly you could also opt for using `Storage::get('path-to/cloudfront/file.svg')` if possible.

**Comment by dmason30** [comment]
Indeed this approach worked for me. Thanks! Adding screenshot just incase tweet disappears for future viewers.

<img width="601" alt="image" src="https://user-images.githubusercontent.com/20278756/164019190-bdaa986f-23e2-4728-a9f0-0c503f38f96b.png">

**Comment by Senexis** [comment, marked as answer]
As an alternative, you can move or copy the logo to the `resources` folder (eg `resources/images/logo.svg`), then simply do:

```php
// config/nova.php
'logo' => realpath(__DIR__.'/../resources/images/logo.svg'),
```

This will work on Vapor just fine.

**Reply by dmason30** [reply]
Yeah this is what i ended up going with, meant to update it here 👍 

`resource_path` also works:

```php
// config/nova.php
'logo' => resource_path('images/logo.svg'),
```

**Comment by eugenefvdm** [comment]
Does it have to be an SVG file?

**Reply by dmason30** [reply]
Yes. The contents of the file is read an injected into the page.

**Comment by emilianotisato** [comment]
For anyone still struggling with this on nova 4.
What I did was create a custom asset (https://nova.laravel.com/docs/4.0/customization/assets.html#defining-assets)
And you override the `AppLogo` vue component. So you have it totally under your control.
<img width="1600" alt="Screenshot 2023-05-16 at 9 44 43" src="https://github.com/laravel/nova-issues/assets/18596215/4226828c-411c-431b-ac8b-2b51c8dfea8c">

**Reply by dotsoftwarenl** [reply]
> For anyone still struggling with this on nova 4. What I did was create a custom asset (https://nova.laravel.com/docs/4.0/customization/assets.html#defining-assets) And you override the `AppLogo` vue component. So you have it totally under your control. <img alt="Screenshot 2023-05-16 at 9 44 43" width="1600" src="https://user-images.githubusercontent.com/18596215/238699928-4226828c-411c-431b-ac8b-2b51c8dfea8c.png">

Hi, I am trying to reproduce this but it seems to not work. Did you edit something elsewhere that you did not mention here?

<img width="725" alt="Screenshot 2023-06-07 at 20 54 54" src="https://github.com/laravel/nova-issues/assets/101074798/143e85e8-c222-4f17-8c57-b11e23e4155a">
